wsmaster/che-core-api-machine-shared/src/main/java/org/eclipse/che/api/machine/shared/dto/MachineSourceDto.java

```diff
@@ -30,10 +31,33 @@
     MachineSourceDto withType(String type);
 
     @Override
-    @FactoryParameter(obligation = MANDATORY)
+    @FactoryParameter(obligation = OPTIONAL)
```

Comment:
fixed, I've added check in validation of machinesource in WorkspaceValidator implementation (used in factory validation)